src/test/java/net/mcreator/integration/generator/GTEnchantmentEffectBlocks.java

```diff
@@ -0,0 +1,135 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.integration.generator;
+
+import net.mcreator.blockly.IBlockGenerator;
+import net.mcreator.blockly.data.BlocklyLoader;
+import net.mcreator.blockly.data.ToolboxBlock;
+import net.mcreator.element.ModElementType;
+import net.mcreator.element.types.Enchantment;
+import net.mcreator.integration.TestWorkspaceDataProvider;
+import net.mcreator.minecraft.ElementUtil;
+import net.mcreator.ui.blockly.BlocklyEditorType;
+import net.mcreator.workspace.Workspace;
+import net.mcreator.workspace.elements.ModElement;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Random;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class GTEnchantmentEffectBlocks {
+
+	public static void runTest(Logger LOG, String generatorName, Random random, Workspace workspace) {
+		Set<String> generatorBlocks = workspace.getGeneratorStats()
+				.getBlocklyBlocks(BlocklyEditorType.ENCHANTMENT_EFFECTS);
+
+		for (ToolboxBlock effectBlock : BlocklyLoader.INSTANCE.getBlockLoader(BlocklyEditorType.ENCHANTMENT_EFFECTS)
+				.getDefinedBlocks().values()) {
+			StringBuilder additionalXML = new StringBuilder();
+
+			if (!BlocklyTestUtil.validateToolboxBlock(effectBlock, generatorBlocks, workspace))
+				continue; // block is not supported by this generator
+
+			if (!BlocklyTestUtil.validateInputs(effectBlock))
+				continue; // failed to validate inputs
+
+			if (!BlocklyTestUtil.populateFields(effectBlock, workspace, random, additionalXML))
+				continue; // failed to populate all fields
+
+			ModElement modElement = new ModElement(workspace,
+					"TestEnchantmentEffectBlock" + effectBlock.getMachineName(), ModElementType.ENCHANTMENT);
+
+			String testXML = effectBlock.getToolboxTestXML();
+
+			// Set block selectors to some value
+			testXML = testXML.replace(
+					"<block type=\"blockstate_selector\"><mutation inputs=\"0\"/><field name=\"block\"></field></block>",
+					"<block type=\"blockstate_selector\"><mutation inputs=\"0\"/><field name=\"block\">"
+							+ TestWorkspaceDataProvider.getRandomMCItem(random,
+							ElementUtil.loadBlocks(modElement.getWorkspace())).getName() + "</field></block>");
+
+			testXML = testXML.replace("<block type=\"" + effectBlock.getMachineName() + "\">",
+					"<block type=\"" + effectBlock.getMachineName() + "\">" + additionalXML);
+
+			Enchantment enchantment = new Enchantment(modElement);
+			enchantment.name = modElement.getName();
+			enchantment.weight = 3;
+			enchantment.anvilCost = 2;
+			enchantment.maxLevel = 3;
+
+			if (effectBlock.getType() == IBlockGenerator.BlockType.PROCEDURAL) {
+				switch (effectBlock.getPreviousStatementConnectionType()) {
+				case "EnchantmentComponent" -> """

```

Comment:
I am not sure if this is right. The strings don't seem to be assigned to the testXML so they just run as statement and are lost (unless I missed something).

I usually test tests by breaking stuff (in this case e.g. JSON syntax) in all relavant blocks and then check if tests catch them for all eligible test candidates